### ISSUE TITLE
Druid powershifting

### ISSUE THREAD

**Issue by Johanes022** [dalaranwow/dalaran-wow#5149]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Powershifting does not remove slow/root effects from the druid.Rhigt now only shifting out and into form breaks roots/slows but not when you powershift (only puff of smoke appears).
Difference between powershifting and regular shifting is that when doing powershift it doesnt change your action bars unike shifting out and back in.
It didnt work for quite a while now, but after you started applying fixes it was working again, now its back as it was before fixes.Strange thing is that noone reported this issue for over 4 years now and its been broken ever since server was open.

**Expected behaviour**: when powershifting it should remove slow/root effects from the druid, currently it does not.Only shifting out and back into form does it.

**Steps to reproduce the problem**:

1. As druid duel any class that can apply slow/root effect .Death knight for example has Chains of ice slow.
2. use any of the following powershifting macros

 /cancelaura Tree of Life(Shapeshift) 
 /cast Tree of Life(Shapeshift)

/cast [stance:1] !Dire Bear Form; [stance:4] !Travel form; [stance:3] !Cat Form; [stance:5] !Tree of Life

/cast [stance:0]!Travel Form;[stance:0/4/6,flyable,nocombat]!Flight Form;[stance:1]!Bear Form;[stance:2]!Aquatic Form;[stance:3]!Cat Form;[stance:4]!Travel Form;[stance:5]!Tree of Life;!Moonkin

3.Be in any form (tree of life for example) when DK applies chains to you.Use powershift macro and notice that you are still beeing slowed by chains.Now apply chains again and shape out and back into form and notice that slow is gone.

**Include proofs for this behaviour**

Powershifting is the act of switching out and quickly or instantly (only a puff of smoke visible) back into Cat Form or Bear Form as a Druid with the Furor Talent to gain an instant boost of Rage or Energy. 
http://wowwiki.wikia.com/wiki/Powershifting

Shapeshifting breaks all movement-impairing effects (like [Slow] or [Hamstring]) and roots ( [Frost Nova] or [Entangling Roots]) except the 3-second daze that can proc when one is attacked from behind by a melee attack.
https://wow.gamepedia.com/Druid#Shapeshifting

**Comment by Zarjofgt** [comment]
It has been reported, and this issue is only with powershifting tree and moonkin form. Possibly the travel forms, havent tested those.

Powershifting also has a major issue where you sometimes count as in no form for a fraction of a second when powershifting.

**Comment by TeridanDW** [comment]
Thank you @Johanes022,
@AttendantTock, could you review this please.

**Comment by AttendantTock** [comment]
Hello,

Apologies, I poorly understood the issue when I tested it in #2698 
I will mark this previous ticket as valid but duplicate of this one, and keep this one open.

This is still valid in the PTR as demonstrated here:
https://youtu.be/Wh1sDdpJWGY

I've tested what I could and it seems to only affect the Boomkin and Tree forms.

@Zarjofgt, could you please detail a bit more the second issue you're writing about?

Thanks.

**Comment by Jebkacik** [comment]
Can I ask, how did you manage to fix this issue?

**Comment by robinsch** [comment]
Like in terms of some insight or a diff?

**Comment by TeridanDW** [comment]
Poke @AttendantTock, could you verify the fix please.

**Comment by AttendantTock** [comment]
<img width="448" alt="capture d ecran 2018-01-31 a 18 38 20" src="https://user-images.githubusercontent.com/29557022/35640761-12454b3e-06b6-11e8-9cca-8a5b8408c9df.png">
 <img width="432" alt="capture d ecran 2018-01-31 a 18 40 09" src="https://user-images.githubusercontent.com/29557022/35640790-3132a190-06b6-11e8-986f-d305829adcad.png">

Hello!

The fix on PTR seems to work fine, both for Tree of Life and Moonkin forms.

***HOWEVER*** the Moonkin aura (5% spell critical chance) is ***NOT*** re-applied by using the /cast !Moonkin Form macro. The Tree of Life aura (+6% heal received) is correctly re-applied.

Regards :)
